st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -502,22 +502,57 @@ def put(self, id, liveaction_api, requester_user, show_secrets=False):
             abort(http_client.INTERNAL_SERVER_ERROR,
                   'Execution object missing link to liveaction %s.' % liveaction_id)
 
-        if liveaction_db.status in LIVEACTION_COMPLETED_STATES:
+        if liveaction_db.status in action_constants.LIVEACTION_COMPLETED_STATES:
             abort(http_client.BAD_REQUEST, 'Execution is already in completed state.')
 
+        if (getattr(liveaction_api, 'result', None) is not None and
+                liveaction_db.status in [
+                    action_constants.LIVEACTION_STATUS_PAUSING,
+                    action_constants.LIVEACTION_STATUS_PAUSED,
+                    action_constants.LIVEACTION_STATUS_RESUMING]):
+            LOG.warning(
+                'The given result attribute of the liveaction update for %s '
+                'is ignored for status change to pause or resume.',
+                liveaction_id
+            )
+
         try:
-            liveaction_db = action_service.update_status(
-                liveaction_db,
-                liveaction_api.status,
-                result=getattr(liveaction_api, 'result', None)
+            if (liveaction_api.status == action_constants.LIVEACTION_STATUS_PAUSING or
+                    liveaction_api.status == action_constants.LIVEACTION_STATUS_PAUSED):
+                liveaction_db, actionexecution_db = action_service.request_pause(
+                    liveaction_db, requester_user.name or cfg.CONF.system_user.user)
+            elif liveaction_api.status == action_constants.LIVEACTION_STATUS_RESUMING:
+                liveaction_db, actionexecution_db = action_service.request_resume(
+                    liveaction_db, requester_user.name or cfg.CONF.system_user.user)
+            else:
+                liveaction_db = action_service.update_status(
+                    liveaction_db,
+                    liveaction_api.status,
+                    result=getattr(liveaction_api, 'result', None)
+                )
+
+                actionexecution_db = ActionExecution.get(liveaction__id=str(liveaction_db.id))
+        except runner_exc.InvalidActionRunnerOperationError as e:
+            LOG.exception('Failed updating liveaction %s. %s', liveaction_db.id, str(e))
+            abort(
+                http_client.BAD_REQUEST,
+                'Failed updating execution because the specific type of request is not supported.'
```

Comment:
Could we return a more specific and friendly error message here?

E.g. Something along the lines of `"PAUSE" operation is not supported for actions with runner type "foo"`.

I know this would probably require changing a messaging of the underlying exception thrown by the container / runner.